content/concept-amendments.md

```diff
@@ -86,13 +86,122 @@ When an amendment gets enabled for the network after the voting process, servers
+
+***TODO: Question: Update the `server_info` doc to include an `amendment_blocked` field descr? https://ripple.com/build/rippled-apis/#server-info. Does this amendment_blocked also display in `server_state`? https://ripple.com/build/rippled-apis/#server-state. If yes, also add descr there?***
+
+To find out which amendments are blocking your `rippled` server, use the [`feature`](reference-rippled.html#feature) admin command and look for features (amendments) that have `"enabled" : true` and `"supported" : false`. These values for a feature mean that the amendment is currently enabled (required) in the latest ledger, but your server does not know how to apply the amendment. For example:
+
+***TODO: Update `features` doc to talk about the implications of when `"enabled" : true` and `"supported" : false`. Useful? https://ripple.com/build/rippled-apis/#feature***
+
+```
+{
+   "id":1,
+   "result":{
+      "features":{
+         "B4D44CC3111ADD964E846FC57760C8B50FFCD5A82C86A72756F6B058DDDF96AD":{
+            "enabled":true,
+            "supported":false,
+            "vetoed":false
+         },
+         "6C92211186613F9647A89DFFBAB8F94C99D4C7E956D495270789128569177DA1":{
+            "enabled":true,
+            "supported":false,
+            "vetoed":false
+         },
+         "B9E739B8296B4A1BB29BE990B17D66E21B62A300A909F25AC55C22D6C72E1F9D":{
+            "enabled":true,
+            "supported":false,
+            "vetoed":false
+         },
+         "1D3463A5891F9E589C5AE839FFAC4A917CE96197098A1EF22304E1BC5B98A454":{
+            "enabled":true,
+            "supported":false,
+            "vetoed":false
+         },
+         "CC5ABAE4F3EC92E94A59B1908C2BE82D2228B6485C00AFF8F22DF930D89C194E":{
+            "enabled":true,
+            "supported":false,
+            "vetoed":false
+         },
+...
+         "08DE7D96082187F6E6578530258C77FAABABE4C20474BDB82F04B021F1A68647":{
+            "enabled":false,
+            "name":"PayChan",
+            "supported":true,
+            "vetoed":false
+         }
+      },
+      "status":"success"
+   }
+}
+```
+
+In this example, conflicts with the following features are causing the `rippled` server to be amendment blocked:
+
+* `B4D44CC3111ADD964E846FC57760C8B50FFCD5A82C86A72756F6B058DDDF96AD`
+
+* `6C92211186613F9647A89DFFBAB8F94C99D4C7E956D495270789128569177DA1`
+
+* `B9E739B8296B4A1BB29BE990B17D66E21B62A300A909F25AC55C22D6C72E1F9D`
+
+* `1D3463A5891F9E589C5AE839FFAC4A917CE96197098A1EF22304E1BC5B98A454`
+
+* `CC5ABAE4F3EC92E94A59B1908C2BE82D2228B6485C00AFF8F22DF930D89C194E`
+
+***TODO: Still need to build out an explanation of how these features became the ones that are amendment blocking the server. Working on amendment blocking my rippled server.***
+
+[Upgrade to a new `rippled` version](tutorial-rippled-setup.html#updating-rippled) to unblock your server and enable it to sync with the network again. ***TODO: Question: Do these instructions need a refresh? Need to take a look.***
```

Comment:
They shouldn't. Even if they do, that's a separate task.

Reply:
You always want to upgrade to latest version, right? Would there be a case where you _don't_ want to use a certain amendment? (I can't think of why you'd want to do that, just asking.)

Reply:
It's certainly not guaranteed that you would want to use a certain amendment. Amendments can change _anything_ about the ledger's transaction processing, so the changes could be distasteful or unprofitable for you. Alternatively, you may find a bug in an amendment's code.

If the rest of the network approves an amendment, your only choices are to accept that and move on, or fork away from the network from a point before the amendment became enabled. Something akin to the latter is what created the Bitcoin Cash and Ethereum Classic forks. (Bitcoin and Ethereum protocols don't have a formalized amendment process in the way the XRP Ledger does, but that doesn't stop them from changing their rules. It just means that changes to their behavior are far less predictable and servers that aren't upgraded automatically fork off rather than becoming amendment blocked).

Reply:
So that means it sort of doesn't matter _which_ amendments aren't supported, right? If _any_ of them are `supported: false`, you have to upgrade to the latest `rippled` version. Else, the decision point described. 

I guess I'm pointing this out because the way the amendments are specified now sort of makes it seem like you could pick and choose which ones you want to support (emphasis on "sort of"). That is, in a way it doesn't matter which specific ones are causing the block; you've got to take all of them if you want to get unblocked.

Reply:
If any amendments are both enabled _and_ unsupported, you must upgrade or fork.

For amendments that are unsupported and _not_ enabled, you can "veto" them (configure your server not to vote in favor of them) and hope that other trusted validators do the same. You're not amendment blocked until those amendments become enabled.

Reply:
@mDuo13 and @ryangyoung - Yes, thank you for this thread! 🙏  This is exactly what I was trying to suss out in trying to amendment block my server -- and then I was going to see what it takes to get it out of that state. 

Got it - so if you are amendment blocked, it may be interesting to know which features are blocking you -- but regardless, you cannot cherry pick which amendments ("enabled" and "unsupported") you want to take to unblock yourself. You just need to upgrade to the version of `rippled` that will unblock you. 

And just to be sure that I fully understand -- this may not always be the latest version of `rippled`, is that right? Meaning, can there be a `rippled` version that does not amendment block previous versions? If this is true, you could have a scenario in which upgrading to a version before the latest version can unblock you. Do we think there are folks operating servers who would have a use case for needing to know how to do this? Meaning, per the doc, the requirement would be to upgrade to the version that will unblock you (I can write a doc on how to determine this), but the recommendation would be to just upgrade to the latest version. 

For amendments that are "not enabled" and "unsupported", you do still have a choice in terms of being able to veto specific amendments, but this discussion is not about getting out of an amendment blocked state, but rather is important in the context of voting, correct?

Reply:
> Can there be a rippled version that does not amendment block previous versions?

Yes, absolutely. I mean, new versions themselves don't automatically amendment-block previous versions. Different versions of `rippled` can coexist as long as no enabled amendments are only known by some versions.

> Do we think there are folks operating servers who would have a use case for needing to know how to [upgrade to not the newest version]?

Some people like to thoroughly read the code or wait for others to test a new version.  Those people may end up a version behind everyone else, which is probably OK depending on the version and why it was released. (If it has security fixes, they should really upgrade ASAP.)

> Meaning, per the doc, the requirement would be to upgrade to the version that will unblock you (I can write a doc on how to determine this), but the recommendation would be to just upgrade to the latest version.

Yes.

> For amendments that are "not enabled" and "unsupported", you do still have a choice in terms of being able to veto specific amendments, but this discussion is not about getting out of an amendment blocked state, but rather is important in the context of voting, correct?

Yes. This is described in the [amendment voting documentation](https://ripple.com/build/amendments/#amendment-voting). To "veto" an amendment means to configure your server not to vote in favor of that amendment. (Anyone not voting in favor of an amendment is effectively voting against it.)